crates/wascc-provider/src/lib.rs

```diff
@@ -437,21 +491,66 @@ struct VolumeBinding {
 fn wascc_run_http(
     host: Arc<Mutex<WasccHost>>,
     data: Vec<u8>,
-    env: EnvVars,
+    _env: EnvVars,
     volumes: Vec<VolumeBinding>,
     log_path: &Path,
     status_recv: Receiver<ContainerStatus>,
+    port_assigned: i32,
 ) -> anyhow::Result<ContainerHandle<ActorHandle, LogHandleFactory>> {
     let mut caps: Vec<Capability> = Vec::new();
 
+    let mut env_map: HashMap<String, String> = HashMap::new();
+    env_map.insert("PORT".to_string(), port_assigned.to_string());
     caps.push(Capability {
         name: HTTP_CAPABILITY,
         binding: None,
-        env,
+        env: env_map, //replaced _env
     });
     wascc_run(host, data, &mut caps, volumes, log_path, status_recv)
 }
 
+#[derive(Debug)]
+struct PortAllocationError {}
+
+impl PortAllocationError {
+    fn new() -> PortAllocationError {
+        PortAllocationError {}
+    }
+}
+impl fmt::Display for PortAllocationError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "all ports are currently in use")
+    }
+}
+impl Error for PortAllocationError {
+    fn description(&self) -> &str {
+        "all ports are currently in use"
+    }
+}
+
+fn find_available_port(
+    port_set: &Arc<Mutex<HashMap<i32, String>>>,
+    pod_name: String,
+) -> Result<i32, PortAllocationError> {
+    let mut range = rand::thread_rng();
+    let mut port: i32 = -1;
```

Comment:
Nit: It would probably be more rusty to have this be an `Option<i32>` and then just do `port.replace` when we find a port

Reply:
Good call. Then we could change the logic on line 548 to `if port.is_none()` and line 551 to `Ok(port.unwrap())`.

https://doc.rust-lang.org/std/option/enum.Option.html

Reply:
To be even safer, we can just do `port.ok_or(PortAllocationError::new())` at the end to transform it into a `Result`